sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/types/PhysicalDataType.scala

```diff
@@ -19,48 +19,83 @@ package org.apache.spark.sql.catalyst.types
 
 import org.apache.spark.sql.types._
 
-sealed abstract class PhysicalDataType
-
-case class PhysicalArrayType(elementType: DataType, containsNull: Boolean) extends PhysicalDataType
-
-class PhysicalBinaryType() extends PhysicalDataType
+sealed abstract class PhysicalDataType{
+  private[sql] def isPrimitive: Boolean
+}
+
+case class PhysicalArrayType(elementType: DataType, containsNull: Boolean)
+  extends PhysicalDataType {
+  override def isPrimitive: Boolean = false
+}
+
+class PhysicalBinaryType() extends PhysicalDataType {
+  override def isPrimitive: Boolean = false
+}
 case object PhysicalBinaryType extends PhysicalBinaryType
 
-class PhysicalBooleanType() extends PhysicalDataType
+class PhysicalBooleanType() extends PhysicalDataType {
+  override def isPrimitive: Boolean = true
```

Comment:
looking at the code changes here, seems it's simpler to add a new trait `PhysicalPrimitiveType` instead of a `def isPrimitive`. Then the caller side can just do `pdt instanceof PhysicalPrimitiveType`

Reply:
Thank you for the code review comments,  I will try to modify the code as you say.

Reply:
> looking at the code changes here, seems it's simpler to add a new trait `PhysicalPrimitiveType` instead of a `def isPrimitive`. Then the caller side can just do `pdt instanceof PhysicalPrimitiveType`

Done

Reply:
> looking at the code changes here, seems it's simpler to add a new trait `PhysicalPrimitiveType` instead of a `def isPrimitive`. Then the caller side can just do `pdt instanceof PhysicalPrimitiveType`

I applied the suggested code changes. Can you please do another review？